bazel/extension_config/extensions_build_config.bzl

```diff
@@ -35,6 +35,8 @@ ENVOY_EXTENSIONS = {
     "envoy.compression.gzip.decompressor":              "//source/extensions/compression/gzip/decompressor:config",
     "envoy.compression.brotli.compressor":              "//source/extensions/compression/brotli/compressor:config",
     "envoy.compression.brotli.decompressor":            "//source/extensions/compression/brotli/decompressor:config",
+    "envoy.compression.zstd.compressor":                "//source/extensions/compression/zstd/compressor:config",
```

Comment:
Why do we need to backport this? It doesn't seem to be a bug fix.

Reply:
it's an core filter, should always enabled, see more https://istio.slack.com/archives/C37A4KAAD/p1701090874408349

Reply:
I see, thanks